service/worker/deletenamespace/deleteexecutions/activities.go

```diff
@@ -175,11 +196,14 @@ func (a *Activities) DeleteExecutionsActivity(ctx context.Context, params Delete
 			metrics.DeleteExecutionFailuresCount.With(a.metricsHandler).Record(1)
 			a.logger.Error("Unable to delete workflow execution.", tag.WorkflowNamespace(params.Namespace.String()), tag.WorkflowID(execution.Execution.GetWorkflowId()), tag.WorkflowRunID(execution.Execution.GetRunId()), tag.Error(err))
 		}
-		activity.RecordHeartbeat(ctx, result)
 		select {
+		case progressCh <- result:
+			// Send the current result to heartbeat go routine.
 		case <-ctx.Done():
+			// Stop deletion on cancellation.
 			return result, ctx.Err()
 		default:
+			// Don't block deletion if a progress channel is full.
```

Comment:
Are you sure that's desirable, probably better to ensure that you can keep heartbeating.
Maybe just give the channel a larger buffer size? Or even just use an atomic pointer instead of a channel to set the latest result.

You could create a background context `WithCancel` for the heartbeating goroutine and `defer cancel()` instead of using this channel.

Reply:
I prefer to use channel here. And yes, I believe I don't want to block deletion, if for some reason heartbeat go routine is blocked. In fact, this only can be if `activity.RecordHeartbeat()` call took too long. Progress won't be lost as it is accumulated in the main goroutine.